src/AsyncProducers.cpp

```diff
@@ -623,7 +679,12 @@ class TightenForkNodes : public IRMutator {
 
     Stmt visit(const LetStmt *op) override {
         Stmt body = mutate(op->body);
-        if (in_fork && !stmt_uses_var(body, op->name)) {
+        // Special case handling for .min & .stride fields.
+        // Atomics generate these let variables for the mutexes
+        // before the storage flatten stage insert the usages of them.
+        bool is_storage_var = op->name.find(".min.") != std::string::npos ||
```

Comment:
This string-matching is a red flag. If the mutex buffers are injected later in lowering, could the modifications to AsyncProducers go away or at least be simpler?

Reply:
This goes away now.